fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java

```diff
@@ -0,0 +1,264 @@
+        hdfsScanRange.setPaimon_split_info(encodeObjectToString(split));
+        // TODO(wangriyu): file length
+        hdfsScanRange.setFile_length(-1);
+
+        TScanRange scanRange = new TScanRange();
+        scanRange.setHdfs_scan_range(hdfsScanRange);
+        scanRangeLocations.setScan_range(scanRange);
+
+        TScanRangeLocation scanRangeLocation = new TScanRangeLocation(new TNetworkAddress("-1", -1));
+        scanRangeLocations.addToLocations(scanRangeLocation);
+
+        scanRangeLocationsList.add(scanRangeLocations);
+    }
+
+    private long nextPartitionId() {
+        return partitionIdGen.getAndIncrement();
+    }
+
+    @Override
+    protected String getNodeExplainString(String prefix, TExplainLevel detailLevel) {
+        StringBuilder output = new StringBuilder();
+
+        output.append(prefix).append("TABLE: ").append(paimonTable.getName()).append("\n");
+
+        if (null != sortColumn) {
+            output.append(prefix).append("SORT COLUMN: ").append(sortColumn).append("\n");
+        }
+        if (!scanNodePredicates.getPartitionConjuncts().isEmpty()) {
+            output.append(prefix).append("PARTITION PREDICATES: ").append(
+                    getExplainString(scanNodePredicates.getPartitionConjuncts())).append("\n");
+        }
+        if (!scanNodePredicates.getNonPartitionConjuncts().isEmpty()) {
+            output.append(prefix).append("NON-PARTITION PREDICATES: ").append(
+                    getExplainString(scanNodePredicates.getNonPartitionConjuncts())).append("\n");
+        }
+        if (!scanNodePredicates.getNoEvalPartitionConjuncts().isEmpty()) {
+            output.append(prefix).append("NO EVAL-PARTITION PREDICATES: ").append(
+                    getExplainString(scanNodePredicates.getNoEvalPartitionConjuncts())).append("\n");
+        }
+        if (!scanNodePredicates.getMinMaxConjuncts().isEmpty()) {
+            output.append(prefix).append("MIN/MAX PREDICATES: ").append(
+                    getExplainString(scanNodePredicates.getMinMaxConjuncts())).append("\n");
+        }
+
+        List<String> partitionNames = GlobalStateMgr.getCurrentState().getMetadataMgr().listPartitionNames(
+                paimonTable.getCatalogName(), paimonTable.getDbName(), paimonTable.getTableName());
+
+        output.append(prefix).append(
+                String.format("partitions=%s/%s", scanNodePredicates.getSelectedPartitionIds().size(),
+                        partitionNames.size() == 0 ? 1 : partitionNames.size()));
+        output.append("\n");
+
+        // TODO: support it in verbose
+        if (detailLevel != VERBOSE) {
+            output.append(prefix).append(String.format("cardinality=%s", cardinality));
+            output.append("\n");
+        }
+
+        output.append(prefix).append(String.format("avgRowSize=%s", avgRowSize));
+        output.append("\n");
+
+        output.append(prefix).append(String.format("numNodes=%s", numNodes));
```

Comment:
build failed in this line

Reply:
what is the error message?

All PR checks passed before the code was merged and why is this so..